src/TwigComponent/doc/index.rst

```diff
@@ -1058,6 +1058,141 @@ Exclude specific attributes:
       My Component!
     </div>
 
+Component with Complex Variants (CVA)
+-------------------------------------
+
+CVA (Class Variant Authority) is a concept from the JS world (https://cva.style/docs/getting-started/variants).
+It's a concept used by the famous shadcn/ui library (https://ui.shadcn.com).
+CVA allows you to display a component with different variants (color, size, etc.),
+to create highly reusable and customizable components.
+You can use the cva function to define variants for your component.
+The first argument of the cva function is the base classes, the classes present in all variants. Then the second
+argument is an array of variants.
+
+.. code-block:: html+twig
+
+    {# templates/components/Alert.html.twig #}
+    {% props color = 'blue', size = 'md' %}
+
+     {% set alert = cva('alert ' ~ attributes.render('class'), {
+        color: {
+            blue: 'bg-blue',
+            red: 'bg-red',
+            green: 'bg-green',
+        },
+        size: {
+            sm: 'text-sm',
+            md: 'text-md',
+            lg: 'text-lg',
+        }
+    }) %}
+
+    <div class="{{ alert.apply({color, size}) }}">
+         {% block content %}{% endblock %}
+    </div>
+
+
+    {# index.html.twig #}
+
+    <twig:Alert color="red" size="lg">
+        <div>My content</div>
+    </twig:Alert>
+    // class="alert bg-red text-lg"
+
+    <twig:Alert color="green" size="sm">
+        <div>My content</div>
+    </twig:Alert>
+    // class="alert bg-green text-sm"
+
+    <twig:Alert class="flex items-center justify-center">
+        <div>My content</div>
+    </twig:Alert>
+    // class="alert bg-blue text-md flex items-center justify-center"
+
+CVA and Tailwind CSS
+~~~~~~~~~~~~~~~~~~~~
+
+CVA work perfectly with tailwindcss. The only drawback is you can have class conflicts,
+to have a better control you can use this following bundle (
+https://github.com/tales-from-a-dev/twig-tailwind-extra
+) in addition to the cva function:
+
+.. code-block:: terminal
+
+    $ composer require tales-from-a-dev/twig-tailwind-extra
+
+.. code-block:: html+twig
+
+    {# templates/components/Alert.html.twig #}
+    {% props color = 'blue', size = 'md' %}
+
+     {% set alert = cva('alert ' ~ attributes.render('class'), {
+        color: {
+            blue: 'bg-blue',
+            red: 'bg-red',
+            green: 'bg-green',
+        },
+        size: {
+            sm: 'text-sm',
+            md: 'text-md',
+            lg: 'text-lg',
+        }
+    }) %}
+
+    <div class="{{ alert.apply({color, size}) | tailwind_merge }}">
```

Comment:
```suggestion
    <div class="{{ alert.apply([color, size], attributes.render('class')) | tailwind_merge }}">
```

1. Should the first argument for apply be a list and not a hash?
2. This ensures the passed classes are always last - this is the common behaviour, correct?

Examples below should be updated similarly.

Reply:
It can't be a list because we use the key to find the variant. Do you see what I mean?

Reply:
Hmm, what does `{color, size}` convert to in php? `['color' => null, 'size' => null]`?

Reply:
Yes exactly. I didn't know that, but Ryan showed me the trick

Reply:
Oh wait, `{color, size}` actually converts to `['color' => 'blue', 'size' => 'md']` in the context of the example in the PR description. Am I correct here?